src/config.rs

```diff
@@ -311,6 +311,16 @@ impl ConfigBuilder {
             }
         }
 
+        let smtp_username = env::var("SENDGRID_USERNAME");
+        let smtp_password = env::var("SENDGRID_PASSWORD");
+        if smtp_username.is_ok() && smtp_password.is_ok() {
```

Comment:
I forgot to add here, you can probably do something like this:

```rust
let sendgrid_creds = (env::var("SENDGRID_USERNAME"), env::var("SENDGRID_PASSWORD"));
if let (Ok(smtp_username), Ok(smtp_password)) = sendgrid_creds {
```

Here, `sendgrid_creds` is a tuple of two, which the `if let` then matches only if both are `Ok`, unpacking the contents in separate vars. You can then lose the `unwraps`.

Could even do it on a single line, but that's probably too long.